nntrainer/graph/network_graph.cpp

```diff
@@ -681,36 +681,57 @@ void NetworkGraph::inPlaceOptimize(Manager &manager) {
   }
 }
 
-void NetworkGraph::init2runContext(InitLayerContext &init_context,
-                                   RunLayerContext &run_context) {
-  // NOTE: this just create the wrappers and does not actually memory inside
-  // these wrappers
-  // TODO: create wrappers for weights - initialize is already done, so the
-  // object creation can be done
-  // TODO: create wrapper for the temporary tensors
-  // TODO: create wrappers for outputs
-  // TODO: create a new context with these new wrappers and then copy assign
-  // run_context
+std::vector<Var_Grad *>
+NetworkGraph::updateRunContext(std::shared_ptr<Manager> &manager,
+                               const std::shared_ptr<LayerNode> &lnode,
+                               const std::vector<Var_Grad *> &prev_inputs) {
+  /**
+   * using copy assignment allows setting run_context without adding more
+   * interfaces
+   */
+  const GraphNode &gnode = *lnode.get();
+  const InitLayerContext &init_context = lnode->getInitContext();
+  std::vector<Var_Grad *> inputs = prev_inputs;
+  if (inputs.empty())
+    inputs = manager->requestInputs(gnode, init_context.getInputDimensions());
+
+  const std::vector<Var_Grad *> &outputs =
+    manager->requestOutputs(gnode, init_context.getOutputDimensions());
+
+  /**
+   * @todo must use existing properties like name/trainable of run_context to
+   * create the new run_context
+   */
+  // const RunLayerContext &run_context = lnode->getRunContext();
+  lnode->updateRunContext(RunLayerContext(
+    manager->requestWeights(gnode, init_context.getWeightsSpec()), inputs,
+    outputs, manager->requestTensors(gnode, init_context.getTensorsSpec())));
+
+  return outputs;
 }
 
 int NetworkGraph::initialize(std::shared_ptr<Manager> manager) {
   int status = ML_ERROR_NONE;
-  // TODO: don't delete adj list - use it here and make this more cleaner/faster
+  /** this contains the map from name to input tensors for each node */
+  std::unordered_map<std::string, std::vector<Var_Grad *>> input_map;
 
   for (unsigned int idx = 0; idx < graph.size(); ++idx) {
-    bool first = idx == 0;
     auto const &lnode = getSortedLayerNode(idx);
-    auto &lptr = lnode->getObject();
+    std::string cur_type = lnode->getType();
     ml_logd("layer name : %s", lnode->getName().c_str());
-    std::string cur_type = lptr->getType();
+
+#if !LAYER_V2
+    auto &lptr = lnode->getObject();
+#endif
 
     /**
      * Set input dimension for all the layers.
      * For input layer, as input dimension is known, set input tensor.
      */
-    if (!first) {
+    if (cur_type != InputLayer::type && idx != 0) {
       std::string l_pre_type = getSortedLayerNode(idx - 1)->getType();
```

Comment:
how about making this to a lambda? this logic is used in number of places `cur_type != InputLayer::type && idx != 0` but likely to change sometime